datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -391,28 +502,50 @@ fn rewrite_column_expr(
 fn build_predicate_expression(
     expr: &Expr,
     schema: &Schema,
-    stat_column_req: &mut Vec<(String, StatisticsType, Field)>,
+    required_columns: &mut RequiredStatColumns,
 ) -> Result<Expr> {
     use crate::logical_plan;
+
+    // Returned for unsupported expressions. Such expressions are
+    // converted to TRUE. This can still be useful when multiple
+    // conditions are joined using AND such as: column > 10 AND TRUE
+    let unhandled = logical_plan::lit(true);
+
     // predicate expression can only be a binary expression
     let (left, op, right) = match expr {
         Expr::BinaryExpr { left, op, right } => (left, *op, right),
+        Expr::Column(name) => {
+            if let Some(expr) =
+                build_single_column_expr(&name, schema, required_columns, false)
```

Comment:
This kind of pattern probably can be written a bit shorter with some combinators. Something like:

`build_single_column_expr(&name, schema, required_columns).ok().or(Ok(unhandled))`

Reply:
An excellent idea -- I will do so